cf-agent/files_editline.c

```diff
@@ -1441,27 +1444,33 @@ static int MatchPolicy(EvalContext *ctx, const char *camel, const char *haystack
             }
 
             if (!escaped)
-            {    
-            // Need to escape the original string once here in case it contains regex chars when non-exact match
-            EscapeRegexChars(ip->name, final, CF_BUFSIZE - 1);
-            escaped = true;
-            }
-            
-            if (opt == INSERT_MATCH_TYPE_IGNORE_EMBEDDED)
             {
-                memset(work, 0, CF_BUFSIZE);
-
-                // Strip initial and final first
-
-                for (firstchar = final; isspace((int)*firstchar); firstchar++)
+                // Need to escape the original string once here in case it contains regex chars when non-exact match
+                // Check size of escaped string, and realloc if necessary
+                size_t escape_regex_len = EscapeRegexCharsLen(ip->name);
+                if (escape_regex_len + 1 > final_size)
                 {
+                    final = xrealloc(final, escape_regex_len + 1);
+                    final_size = escape_regex_len;
```

Comment:
You're not being consistent here. Earlier you did:

<pre>
final = xstrdup(ip->name);
final_size = strlen(final) + 1;
</pre>

..which would make ``final_size`` exactly the size of the buffer, including the '\0', but here you assign it the length of the string. Should probably be changed to ``final_size = escape_regex_len + 1;``.